### PR TITLE
Migrate to pnpm

### PR DESCRIPTION
**Pull request by leoeuclids**
Migrates to pnpm
- [x] Package.json
- [x] CI 
- [x] Upgrade to `ember-cli-fastboot@4`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,25 +19,20 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-      - name: Install Node
-        uses: actions/setup-node@v3
+      - uses: wyvox/action-setup-pnpm@v2
```
Comment (SergeAstapov) [inline .github/workflows/ci.yml:22]: minor, this was removed from addon blueprint recently https://github.com/embroider-build/addon-blueprint/blob/main/files/.github/workflows/ci.yml#L22

File: .github/workflows/publish-unstable.yml [hunk 2]
```diff
@@ -16,17 +16,12 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-      - name: Install Node
-        uses: actions/setup-node@v3
+      - uses: wyvox/action-setup-pnpm@v2
```
Comment (SergeAstapov) [inline .github/workflows/publish-unstable.yml:19]: note for next step - replace this with `put-built-npm-package-contents-on-branch` per https://github.com/embroider-build/addon-blueprint/blob/main/files/.github/workflows/push-dist.yml

File: package.json [hunk 3]
```diff
@@ -5,21 +5,15 @@
...
+    "prepare": "cd addon && pnpm build",
+    "test": "cd test-app && pnpm test"
   },
   "devDependencies": {
     "release-plan": "^0.8.0"
   },
   "volta": {
-    "node": "16.20.0",
-    "yarn": "1.22.18"
+    "node": "16.20.0"
```
Comment (SergeAstapov) [inline package.json:16]: as we bump for CI, worth to bump for development as well?

Comment (leoeuclids) [reply package.json]: Bumped

File: .github/workflows/publish-unstable.yml (hunk 2 above)
Comment (leoeuclids) [reply .github/workflows/publish-unstable.yml]: I've swapped the `wyvox/action-setup-pnpm@v2` for the `pnpm/action-setup@v3`, as per the blueprint, but I'm not sure how to incorporate the `put-built-npm-package-contents-on-branch` in it. Should it just be added as the next step after `Install dependecies`?

Comment (SergeAstapov) [reply .github/workflows/publish-unstable.yml]: @leoeuclids we basically should update this whole file to match https://github.com/embroider-build/addon-blueprint/blob/main/files/.github/workflows/push-dist.yml (replacing placeholders and if/else branching of course) we can do that in follow up PR

**Review by SergeAstapov — APPROVED** [review]

**Comment by SergeAstapov** [comment]
Thank you @leoeuclids!

**Comment by leoeuclids** [comment]
Thank you @SergeAstapov!

**Pull request merged**